explainer/Explainer.md

```diff
@@ -1,9 +1,369 @@
+of participants and are collected in the 
+[WoT Use Cases and Requirements document](https://w3c.github.io/wot-usecases/).
+In the following we discuss use cases and 
+extract a few of the most critical requirements
+driving the design of WoT Discovery.
+
+### Use Case: Integration
+It is often the case in IoT systems that devices provided by different manufacturers
+need to be integrated.  For example, consider a Retail use case
+where a door sensor, a temperature sensor, and an RGB light need to be integrated
+with an automation to indicate when a freezer door has been left open or the 
+temperature is out of range.  As another example, a user might want to integrate an
+accessibility control, such as a blow switch, with an IoT device that operates a 
+light or calls an elevator.  In these cases, it is not enough to have a common
+interface description format, which the WoT TD provides.  We also need a standard
+way to find and fetch the TDs for specific devices so that they can be integrated.
+
+Integration can take place during development time where a developer may
+use Thing Models to understand the behavior of classes of devices and
+build automation rules that will combine instances of these classes to create
+new IoT services.  However, at installation time, to apply such automation
+rules, it is necessary to search for actual instances of Thing Descriptions 
+(representing installed Thing instances) satisfying the requirements of the automations.
+
+Discovery currently targets the second half of this problem, finding Thing
+instances satisfying a set of constraints.  Managing Thing Models during
+development time is currently out of scope.
+
+### Use Case: Public IoT Services
+In a Smart City, we may want to provide access to IoT devices as a public service,
+for example to report local air quality conditions.  In Retail, we may want to
+make available semi-public services (perhaps requiring pre-registration) to access
+accessiblity devices, such as chair lifts.  In this case, a user wants to discover
+services, probably selected from a large number, that meet some criteria, such
+as location or type.
+
+### Requirements
+WoT Discovery has been designed with the following goals in mind:
+- security and privacy;
+- ease of use;
+- integration with existing ecosystems;
+- both local and global access;
+- scalability to large numbers of Things, including distributing searches over multiple directory services; and
+- support for both self-description and external descriptions (to support brownfield scenarios).
+
+Some of these are in conflict.  For example, while we want to integrate with
+existing ecosystems, we don't want to expose users to known or potential 
+security hazards in those ecosystems.
+Making discovery simple enough to be used on small devices for self-description
+while also secure and
+scalable to a large number of IoT devices is also a significant challenge.
```

Comment:
It is a mild repetition of what has been said in the introduction, but it might not hurt too much.